docusaurus/docs/guides/deployments/10-linux/10-controller/10-deploy.mdx

```diff
@@ -0,0 +1,98 @@
+---
+title: Controller Deployment
+sidebar_label: Controller
+id: deploy
+---
+
+import LinuxPackageRepo from '/docs/_linux-package-repo.mdx';
+import CliLogFormats from '/docs/_cli-log-levels-and-formats.md';
+
+This article is about deploying a Ziti Controller as a Linux system service. [The controller introduction](/learn/introduction/03-components.md#openziti-controller) introduces the architectural concepts used in this guide.
+
+## Installation
+
+The controller package `openziti-controller` provides a systemd service unit and optional bootstrapping. The package depends on the `openziti` package which provides the `ziti` CLI. The easiest way to install both is the cross-platform install script for RPM and Debian distributions.
+
+```text
+curl -sS https://get.openziti.io/install.bash \
+| sudo bash -s openziti-controller
+```
+
+<LinuxPackageRepo />
+
+## Bootstrapping
+
+You can opt-in to generating a PKI, config, and database by installing interactively with prompts or by setting these values.
+
+1. In `/opt/openziti/etc/controller/service.env` (the answer file for the service)
+    1. Set `ZITI_BOOTSTRAP=true`
+1. In `/opt/openziti/etc/controller/bootstrap.env` (the answer file for bootstrapping the controller)
+    1. Set `ZITI_CTRL_ADVERTISED_ADDRESS` to the FQDN of the controller
+    1. Set `ZITI_PWD` to the desired management password for user `admin`. You may delete this after bootstrapping.
+
+### Generating a Public Key Infrastructure
+
+The controller service will generate a root CA and issue an intermediate CA cert during the first startup.
+
+The controller needs a CA to issue certificates for edge identities during enrollment. This CA is known as the edge signer. It's a good idea to secure the root in a different location and use an intermediate CA for this purpose. This allows you to recover in a scenario where the intermediate was lost without changing the root of trust and invalidating all edge enrollments.
+
+Disable bootstrapping a PKI by setting `ZITI_BOOTSTRAP_PKI=false` in `/opt/openziti/etc/controller/service.env`.
+
+Check out [the PKI page](/learn/core-concepts/pki.md) for an overview of these concepts.
+
+### Generating a Configuration
+
+The controller service will generate a configuration during the first startup.

```

Comment:
Implementation question: Can we make the existence of the answer file a requirement here? As in, the user must have placed an answer file into the expected location for the service to startup the first time, so that the proper configuration can be generated? That way the configuration step mentioned before is basically "make the answer file" (exactly how the answer file is created is an implementation detail that doesn't need to be documented). Then the process could remove the answer file for the user so that it's not left behind. That would allow the ZITI_PWD to be put into the answer file as well since the password gets removed by the initialization step.

Then when the controller starts it can verify the answer file exists (or doesn't) to use that to determine if the config should be generated(regenerated) on start.

Reply:
I'm hearing two suggestions: require an answer file when not providing a configuration, and remove the answer file (including the admin password if it was provided there) when the configuration has been generated.

The answer file (`bootstrap.env`) is created without answers (null values) when the controller is installed. If it is missing, then systemd will fail to start the service because this answer file is one of the required env files.

If `bootstrap.env` does not contain the required answers (e.g., `ZITI_CTRL_ADVERTISED_ADDRESS`), then service will not start because bootstrapping will return an error.

The answer file can only be removed by refactoring the service unit and entrypoint script because it is a required env file for the service unit and life cycle scriptlets rely on the service.env and bootstrap.env answer/env files to determine whether or not a particular question is already answered. For example, if bootstrapping is enabled and a question is not answered then the package manager upgrade scriptlet may then prompt for the answer or take background action to correct an ambient kernel capability. So, the answer files are parcel to the package life cycle.

I raised an issue to redress how db bootstrapping works in https://github.com/openziti/ziti/issues/2000 in response to the suggestion that the Linux package should proactively delete the ziti password from disk after the database is initialized.